src/spectests/get_local.rs

```diff
@@ -116,155 +118,131 @@ fn create_module_1() -> ResultObject {
-    };
-    let invoke_fn: fn(i64, &Instance) -> i64 =
-        get_instance_function!(result_object.instance, func_index);
-    let result = invoke_fn(3 as i64, &result_object.instance);
-    assert_eq!(result, 3 as i64);
+    let result = result_object
+        .instance
+        .call("c6_l70_action_invoke", &vec![Val::I64(3 as i64)][..])
+        .expect("Missing result in c6_l70_action_invoke");
+    assert_eq!(result, Some(Val::I64(3 as i64)));
 }
 
 // Line 71
-fn c7_l71_action_invoke(result_object: &ResultObject) {
+fn c7_l71_action_invoke(result_object: &mut ResultObject) {
     println!("Executing function {}", "c7_l71_action_invoke");
-    let func_index = match result_object.module.info.exports.get("type-param-f32") {
-        Some(&Export::Function(index)) => index,
-        _ => panic!("Function not found"),
-    };
-    let invoke_fn: fn(f32, &Instance) -> f32 =
-        get_instance_function!(result_object.instance, func_index);
-    let result = invoke_fn(4.4 as f32, &result_object.instance);
-    assert_eq!(result, 4.4 as f32);
+    let result = result_object
+        .instance
+        .call("c7_l71_action_invoke", &vec![Val::F32(4.4 as u32)][..])
+        .expect("Missing result in c7_l71_action_invoke");
+    assert_eq!(result, Some(Val::F32(4.4 as u32)));
 }
 
 // Line 72
-fn c8_l72_action_invoke(result_object: &ResultObject) {
+fn c8_l72_action_invoke(result_object: &mut ResultObject) {
     println!("Executing function {}", "c8_l72_action_invoke");
-    let func_index = match result_object.module.info.exports.get("type-param-f64") {
-        Some(&Export::Function(index)) => index,
-        _ => panic!("Function not found"),
-    };
-    let invoke_fn: fn(f64, &Instance) -> f64 =
-        get_instance_function!(result_object.instance, func_index);
-    let result = invoke_fn(5.5 as f64, &result_object.instance);
-    assert_eq!(result, 5.5 as f64);
+    let result = result_object
+        .instance
+        .call("c8_l72_action_invoke", &vec![Val::F64(5.5 as u64)][..])
+        .expect("Missing result in c8_l72_action_invoke");
+    assert_eq!(result, Some(Val::F64(5.5 as u64)));
 }
 
 // Line 75
-fn c9_l75_action_invoke(result_object: &ResultObject) {
+fn c9_l75_action_invoke(result_object: &mut ResultObject) {
     println!("Executing function {}", "c9_l75_action_invoke");
-    let func_index = match result_object.module.info.exports.get("type-mixed") {
-        Some(&Export::Function(index)) => index,
-        _ => panic!("Function not found"),
-    };
-    let invoke_fn: fn(i64, f32, f64, i32, i32, &Instance) =
-        get_instance_function!(result_object.instance, func_index);
-    let result = invoke_fn(
-        1 as i64,
-        2.2 as f32,
-        3.3 as f64,
-        4 as i32,
-        5 as i32,
-        &result_object.instance,
-    );
-    assert_eq!(result, ());
+    let result = result_object
+        .instance
+        .call(
+            "c9_l75_action_invoke",
+            &vec![
```

Comment:
You can just do `&[...]` instead of creating a `Vec`.

Reply:
Fixed.